### ISSUE TITLE
Possible issues / inconsistent code in JTable

### ISSUE THREAD

**Issue by joomdonation** [joomla/joomla-cms#6901]
We have several code like this in JTable

````php
if (property_exists($this, 'checked_out') || property_exists($this, 'checked_out_time'))
{
	$query->where('(' . $this->getColumnAlias('checked_out') . ' = 0 OR ' . $this->getColumnAlias('checked_out') . ' = ' . (int) $userId . ')');
	$checkin = true;
}
````

Here is one example https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/table/table.php#L1568-L1572

As you see, we allow developers to set column alias for a column in case they want to use different field name than the one using by Joomla. However, the problem is that in in the check in if command, we hardcode the field name to checked_out and checked_out_time, not using getColumnAlias. So if someone uses column alias, the code won't run properly.

Should we update the code to make it consistent and correct it in case someone uses column alias?

**Comment by julienV** [comment]
it would make sense indeed !<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6901">issues.joomla.org/joomla-cms/6901</a>.</sub>

**Comment by brianteeman** [comment]
Are you going to submit a Pull Request for this<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6901">issues.joomla.org/joomla-cms/6901</a>.</sub>

**Comment by joomdonation** [comment]
Hi Brian

Yes. I will find sometime to do it on tomorrow.

**Comment by joomdonation** [comment]
OK. I created the PR. Please help closing this issue, Brian

**Comment by brianteeman** [comment]
Closed as we have a PR #9009 Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6901">issues.joomla.org/joomla-cms/6901</a>.</sub>
